apps/05Mastermind.js

```diff
@@ -26,18 +25,56 @@ function getRandomInt(min, max) {
   return Math.floor(Math.random() * (max - min)) + min;
 }
 
-function generateHint() {
-    // your code here
+function generateHint(solution, guess) {
+    var solutionArray = solution.split('');
+    var guessArray = guess.split('');
+    var correctLetterLocations = 0;
+    var correctLetters = 0;
+    var i = 0;
+
+    for (i = 0; i < guessArray.length; i++) {
+      if (solutionArray[i] === guessArray[i]) {
+        correctLetterLocations++;
+        solutionArray[i] = null;
+      }
+    }
+
+    for (i = 0; i < guessArray.length; i++) {
+        var targetIndex = solutionArray.indexOf(guessArray[i]);
+          if(targetIndex > -1) {
+            correctLetters++;
+            solutionArray[targetIndex] = null;
+        }
+    }
+
+    return correctLetterLocations + '-' + correctLetters;
 }
 
+
 function mastermind(guess) {
-    // your code here
-}
+  solution = 'abcd';
```

Comment:
Might want to remove this; we'll want to take advantage of the random solution when we run the app with `node`